animal.go

```diff
@@ -109,36 +93,50 @@ func (g game) Play(r io.Reader, w io.Writer) error {
 		}
 
 		switch question {
-		case AnswerWin:
-			fmt.Fprintf(w, "I successfully guessed your animal! Awesome!\n")
+		case "AnswerWin":
+			fmt.Fprintln(w, "I successfully guessed your animal! Awesome!")
+			g.transcript.WriteString("I successfully guessed your animal! Awesome!")
 			g.Running = false
-		case AnswerLose:
+		case "AnswerLose":
 			fmt.Fprintf(w, "You stumped me! Well done!\n")
+			g.transcript.WriteString("You stumped me! Well done!\n")
 			g.LearnNewAnimal(r, w, prev2)
 			g.Running = false
 		}
 	}
 	return nil
 }
 
+func (g game) Transcript() string {
+	return g.transcript.String()
+}
+
 // LearnNewAnimal - Function to learn the question to add for new animal
-func (g game) LearnNewAnimal(r io.Reader, w io.Writer, pq string) {
+func (g *game) LearnNewAnimal(r io.Reader, w io.Writer, pq string) {
 	var input string
 	fmt.Fprintln(w, "Please tell me the animal you were thinking about: ")
+	g.transcript.WriteString("Please tell me the animal you were thinking about: ")
 	scanner := bufio.NewScanner(r)
 	scanner.Scan()
 	input = scanner.Text()
+	g.transcript.WriteString(input + "\n")
 
 	fmt.Fprintf(w, "What would be a Yes/No question to distinguish %s from other animals: ", input)
+	g.transcript.WriteString(fmt.Sprintf("What would be a Yes/No question to distinguish %s from other animals: ", input))
 	scanner.Scan()
 	qDistinctive := scanner.Text()
+	g.transcript.WriteString(qDistinctive + "\n")
 
 	fmt.Fprintf(w, "What would be the answer to the question - \"%s\" for %s: ", qDistinctive, input)
+	g.transcript.WriteString(fmt.Sprintf("What would be the answer to the question - \"%s\" for %s: ", qDistinctive, input))
 
 	scanner.Scan()
-	ans, err2 := GetUserYesOrNo(strings.NewReader(scanner.Text()))
-	if err2 != nil {
-		fmt.Println("error getting ans", err2)
+	ans, err2 := g.GetUserYesOrNo(strings.NewReader(scanner.Text()))
```

Comment:
It's quite idiomatic to re-use the `err` variable many times throughout a function—once we've set it and checked it, we don't use it again, so we can overwrite it with the next error value.

Reply:
updated this.